tests/test_events/handler/addTab.test.js

```diff
@@ -0,0 +1,119 @@
+import fs from "fs";
+import path from "path";
+import untildify from "untildify";
+import {DATETIME_FORMAT, DEFAULT_BRANCH} from "../../../src/constants";
+import {
+    randomBaseRepoPath,
+    randomRepoPath,
+    addUser,
+    Config,
+    getConfigFilePath,
+    DUMMY_FILE_CONTENT
+} from "../../helpers/helpers";
+import dateFormat from "dateformat";
+import {readYML} from "../../../src/utils/common";
+import {VSCODE} from "../../../src/constants";
+import {pathUtils} from "../../../src/utils/path_utils";
+import {tabEventHandler} from "../../../src/events/tab_event_handler";
+import { createSystemDirectories } from "../../../src/utils/setup_utils";
+
+// Helper method that asserts that no tabs were recorded
+const assertNoTabsRecorded = (tabsRepo) => {
+    // Verify no tab file should be generated
+    let tabFiles = fs.readdirSync(tabsRepo);
+    console.log("Tab files:", tabFiles);
+    expect(tabFiles).toHaveLength(0);    
+}
+
+describe("addTab", () => {
+	const repoPath = randomRepoPath();
+    const baseRepoPath = randomBaseRepoPath();
+    const configPath = getConfigFilePath(baseRepoPath);
+    const configUtil = new Config(repoPath, configPath);
+    untildify.mockReturnValue(baseRepoPath);
+
+    const pathUtilsObj = new pathUtils(repoPath, DEFAULT_BRANCH);
+    const tabsRepo = pathUtilsObj.getTabsRepo();
+    const newFilePath = path.join(repoPath, "new.js");
+
+    beforeEach(() => {
+        // Create directories
+        fs.mkdirSync(repoPath, { recursive: true });
+        fs.mkdirSync(tabsRepo, { recursive: true });
+        createSystemDirectories();   
+        console.log(`Creating directories: repoPath=${repoPath}, tabsRepo=${tabsRepo}`);     
+        jest.clearAllMocks();
+        untildify.mockReturnValue(baseRepoPath);
+        const returnedPath = untildify();
+        console.log(`untildify returned: ${returnedPath}`);
+    });
+
+    afterEach(() => {
+        fs.rmSync(repoPath, { recursive: true, force: true });
+        fs.rmSync(baseRepoPath, { recursive: true, force: true });
+    });
+
+    test('untildify should return the correct mocked path', () => {
```

Comment:
Why are we writing a test of a third party library?

Reply:
I had added this while debugging why tests weren't working. Removing this